### PR TITLE
Feature signing service signing ethereum transactions

### PR DESCRIPTION
**Pull request by rwrzesien**
Resolves https://github.com/golemfactory/concent/issues/633.

### CODE REVIEW

File: signing_service/signing_service/signing_service.py [hunk 1]
```diff
@@ -196,25 +370,55 @@ def __call__(self, parser, namespace, values, option_string=None):
...
     ethereum_private_key_parser_group.add_argument(
         '--ethereum-private-key-from-env',
         dest='ethereum_private_key',
-        action=make_secret_provider_factory(env_variable_name='ETHEREUM_PRIVATE_KEY', base64_convert=True),
+        action=make_secret_provider_factory(env_variable_name='ETHEREUM_PRIVATE_KEY'),
         help='Ethereum private key for Singing Service.',
     )
-    parser.set_defaults(ethereum_private_key=b64decode(os.environ.get('ETHEREUM_PRIVATE_KEY')) if os.environ.get('ETHEREUM_PRIVATE_KEY') is not None else None)  # type: ignore
+    parser.set_defaults(
+        ethereum_private_key=os.environ.get('ETHEREUM_PRIVATE_KEY')
```
Comment (dybi) [inline signing_service/signing_service/signing_service.py]: conditional as default doesn't seem right... especially, that `None` seem to be a bad default value for a private key ;)

File: signing_service/signing_service/signing_service.py [hunk 2]
```diff
@@ -196,25 +370,55 @@ def __call__(self, parser, namespace, values, option_string=None):
...
+        help='Singing Service private key.',
+    )
+    signing_service_private_key_parser_group.add_argument(
+        '--signing-service-private-key-from-env',
+        dest='signing_service_private_key',
+        action=make_secret_provider_factory(env_variable_name='SIGNING_SERVICE_PRIVATE_KEY', base64_convert=True),
+        help='Singing Service private key.',
+    )
+    parser.set_defaults(
+        signing_service_private_key=b64decode(os.environ.get('SIGNING_SERVICE_PRIVATE_KEY'))  # type: ignore
```
Comment (dybi) [inline signing_service/signing_service/signing_service.py]: same as above

File: signing_service/signing_service/tests/tests.py [hunk 3]
```diff
@@ -116,6 +131,267 @@ def test_that_signing_service_should_reraise_different_socket_erros(self):
         mock_socket_close.assert_called_once()
 
 
+class SigningServiceHandleConnectionTestCase(SigningServiceIntegrationTestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.host = '127.0.0.1'
+        self.port = 8000
```
Comment (dybi) [inline signing_service/signing_service/tests/tests.py]: port should not be fixed - please use `unused_tcp_port` fixture

File: signing_service/signing_service/signing_service.py [hunk 4]
```diff
@@ -133,10 +244,72 @@ def _validate_arguments(self):
...
+            v        = transaction.v,
+            r        = transaction.r,
+            s        = transaction.s,
+        )
+
 
 def _parse_arguments() -> argparse.Namespace:
     def make_secret_provider_factory(read_command_line=False, env_variable_name=None, use_file=False, base64_convert=False):
         def wrapper(**kwargs):
+            # Remove 'required' as it is added automatically by ArgumentParser.
```
[the remark this answers is not in the document]
Comment (rwrzesien) [reply signing_service/signing_service/signing_service.py]: Removed along with changed from https://github.com/golemfactory/concent/pull/688.

File: signing_service/signing_service/signing_service.py (hunk 1 above)
Comment (rwrzesien) [reply signing_service/signing_service/signing_service.py]: Removed along with changes from https://github.com/golemfactory/concent/pull/688.

File: signing_service/signing_service/signing_service.py (hunk 2 above)
Comment (rwrzesien) [reply signing_service/signing_service/signing_service.py]: Removed along with changes from https://github.com/golemfactory/concent/pull/688.

File: signing_service/signing_service/tests/tests.py (hunk 3 above)
Comment (rwrzesien) [reply signing_service/signing_service/tests/tests.py]: Done in base branch https://github.com/golemfactory/concent/pull/688.

**Comment by rwrzesien** [comment]
Closed by accident, reopened in https://github.com/golemfactory/concent/pull/725.

**Pull request closed without merging**